Add and view quizzes

sofiaedstrom opened the pull request:
* View 10 quizzes at a time (ordered by quiz id)
* Search quizzes by name (<= 10 at a time is retrieved)
* Load more quizzes (10 at a time), both for "all" quizzes and search results
* Delete quiz
* Modal with quiz details is displayed when pressing button "Visa"

DzenoBa committed:
+ quizes page

DzenoBa committed:
+ search quizes

DzenoBa committed:
+ create quiz

DzenoBa committed:
+ load more quizes

sofiaedstrom committed:
Merge branch 'develop' into quizes

sofiaedstrom committed:
Merge branch 'develop' into quizes

sofiaedstrom committed:
indentation + unused imports fix

sofiaedstrom committed:
correct port

sofiaedstrom committed:
remove unused statements

sofiaedstrom committed:
change naming

sofiaedstrom committed:
correct response status

sofiaedstrom committed:
introduce base url

sofiaedstrom committed:
specify request method in getAll

sofiaedstrom committed:
basic delete quiz

sofiaedstrom committed:
split components in files

sofiaedstrom committed:
login utils for testing admin pages

sofiaedstrom committed:
quiz tests

sofiaedstrom committed:
test navigation to quizzes page

sofiaedstrom committed:
admin panel and quizzes page page objects

sofiaedstrom committed:
prepare for delete quiz test

sofiaedstrom committed:
Merge branch 'develop' into quizes

# Conflicts:
#	src/main/js/screens/app/components/GakuseiAdminNav.js

sofiaedstrom committed:
add modal with quiz details

Doverstav commented on src/main/js/screens/app/screens/quizes/QuizBox.js:
```diff
@@ -0,0 +1,106 @@
+    this.openModal = this.openModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+  }
+
```
onInputChange is not used?

sofiaedstrom replied:
✅ removed!

Doverstav commented on src/main/js/screens/app/screens/quizes/QuizBox.js:
```diff
@@ -0,0 +1,106 @@
+      [event.target.name]: event.target.value,
+    });
+  }
+
```
mouseEnter and mouseExit differ in style

sofiaedstrom replied:
✅ uniform bindings fixed

Doverstav commented on src/main/js/screens/app/screens/quizes/QuizBox.js:
```diff
@@ -0,0 +1,106 @@
+            <strong>{quiz.name}</strong> har raderats.
+          </Alert>
+        ) : (
+          <div>
```
This <Panel> could perhaps be made into a separate component?

sofiaedstrom replied:
i think its a bit to small

Doverstav commented on src/main/js/screens/app/screens/quizes/QuizBox.js:
```diff
@@ -0,0 +1,106 @@
+                :
+                null
+              }
+            </Panel>
```
This could perhaps be made into a separate component?

sofiaedstrom replied:
It will probably be separated in the next PR!

Doverstav commented on src/main/js/screens/app/screens/quizes/QuizForm.js:
```diff
@@ -0,0 +1,107 @@
+      });
+    });
+  }
+
```
I like this ⭐️ 

Perhaps, in order to clarify WHEN errorMsg is shown, move condition checking to render, i.e. `{this.state.error != null ? renderErrorMsg : null}` instead of `{renderErrorMsg()}`

sofiaedstrom replied:
good idea!
✅

Doverstav commented on src/main/js/screens/app/screens/quizes/QuizForm.js:
```diff
@@ -0,0 +1,107 @@
+            />
+          </FormGroup>
+          <FormGroup>
+            {this.renderErrorMsg()}
```
See comment on `renderErrorMsg()` for possible rewrite

sofiaedstrom replied:
✅

Doverstav commented on src/main/js/screens/app/screens/quizes/QuizForm.js:
```diff
@@ -0,0 +1,107 @@
+                label="create"
+                bsStyle="primary"
+                name="create"
+                disabled={!this.state.name || !this.state.description}
```
Maybe extract boolean condition to separate function to clarify WHEN button is disabled (thorugh descriptive function name). This is not a terribly complicated expression so its probably fine, but extracting conditions to function could increase readability,

sofiaedstrom replied:
good idea! I already rewrote this in a later PR when dealing with form validation, so won't fix it now

Doverstav commented on src/main/js/screens/app/screens/quizes/QuizesScreen.js:
```diff
@@ -0,0 +1,144 @@
+import QuizBox from './QuizBox';
+import QuizForm from './QuizForm';
+
+class QuizesScreen extends React.Component {
```
QuizzesScreen? 😛

sofiaedstrom replied:
✅ 😂 it was everywhere.

Doverstav commented on src/main/js/screens/app/screens/quizes/QuizesScreen.js:
```diff
@@ -0,0 +1,144 @@
+    }
+  }
+
+  handeLoadMore() {
```
Typo in function name

sofiaedstrom replied:
✅

Doverstav commented on src/main/js/screens/app/screens/quizes/QuizesScreen.js:
```diff
@@ -0,0 +1,144 @@
+            disabled={
+              (this.state.quizes.length - (10 * this.state.offset)) < 10
+            }
+            onClick={() => this.handeLoadMore()}
```
Bound by `() =>` and not in constructor.

sofiaedstrom replied:
✅

Doverstav commented on src/main/js/screens/app/screens/quizes/QuizesScreen.js:
```diff
@@ -0,0 +1,144 @@
+    });
+    this.getQuizes(event.target.value, 0);
+  }
+
```
Are _searchPattern_ and _offset_ needed as parameters? Aren't they accessible in the state?

Doverstav reviewed (commented):
Overall good. Some things that could be discussed:
- Some functions are defined through `() =>` and some are written as "normal" functions. Should we decide on a style and stick to it, or is mixing style ok? If we decide to stick to a certain style, some more changes might be needed.
- When handling server responses, sometimes `switch-case` is used, sometimes `if-else`. Do we need to decide on a style here as well or is a mix ok?

sofiaedstrom committed:
remove unused function onInputChange

sofiaedstrom committed:
fix typo in function name

sofiaedstrom committed:
uniform bindings

sofiaedstrom committed:
update ui descriptions

sofiaedstrom committed:
rewrite renderErrorMsg

sofiaedstrom committed:
fix the quizzes typos

sofiaedstrom committed:
remove unnecessary surrounding block in function

sofiaedstrom committed:
fix linter error

sofiaedstrom committed:
update navbar to include quiz and only show about page when not logged in

sofiaedstrom committed:
switch case -> if else

sofiaedstrom committed:
rewrite alert handling

sofiaedstrom committed:
fix even more quizzes typos

sofiaedstrom committed:
update e2e test page objects according to typo fixes

sofiaedstrom committed:
close browser window after final test case

sofiaedstrom commented:
* updated according to feedback
* cleaned up alert message handling in QuizzesScreen 
* small improvement of error handling in QuizzesScreen
* updated e2e test to match recent changes
